### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by altexy**
@ldeveloperl1985  here is my version with the latest changes.

API is completely changes. now returns complete resty-http response with parsed JSON body if any

### CODE REVIEW

**Comment by altexy** [comment]
#4

File: oxdweb.lua [hunk 1]
```diff
@@ -1,133 +1,91 @@
...
 
-    ngx.log(ngx.DEBUG, "Host: " .. conf.oxd_host .. "/" .. command .. " Request_Body:" .. jsonBody .. " response_body: " .. res.body)
-
-    if pcall(decode, res.body) then
-        return decode(res.body)
-    else
-        return { status = "error", description = "Please see the oxd log" }
+    if err then
+        ngx.log(ngx.ERR, "HTTP error: ", err)
+        return { status = "error", description = "HTTP error: ", err }
```
Comment (ldeveloperl1985) [inline oxdweb.lua]: Need to send here HTTP status code. In successful case, we are sending HTTP Status code with body.

Comment (altexy) [reply oxdweb.lua]: Gotcha

File: oxdweb.lua [hunk 2]
```diff
@@ -1,133 +1,89 @@
...
 
-function _M.uma_rp_get_rpt(conf, token)
-    local commandAsJson = json:encode(conf)
-    local response = _M.execute_http(conf, commandAsJson, token, "uma-rp-get-rpt")
-    return response
-end
+    if res.status < 300 then
+        local json, err = cjson.decode(res.body)
+        if err then
+            ngx.log(ngx.ERR, )
```
Comment (ldeveloperl1985) [inline oxdweb.lua]: Missing error message

**Comment by ldeveloperl1985** [comment]
@altexy everyting is OK now. But you need to solve the conflict so we can merge this pull request.

**Pull request merged**